### PR TITLE
chore(tests): Uniformly use promises rather than done() callback.

### PR DESCRIPTION
**Pull request by rfk**
The API tests were a strange mixture of returning promises and using the done() callback; this makes them all uniformly use promises.

### CODE REVIEW

**Comment by rfk** [comment]
@mozilla/fxa-devs r?

File: test/api.js [hunk 1]
```diff
@@ -750,7 +757,7 @@ describe('/v1', function() {
           });
         }
 
-        it('should get auth token with secret', function(done){
+        it('should get auth token with secret', function(){
           return getCode(clientId).then(function(code) {
```
Comment (shane-tomlinson) [inline test/api.js]: I'm a bit surprised there weren't problems with this test earlier since both a promise are returned and `done` specified.

Comment (rfk) [inline test/api.js]: Yeah me too, the `mocha` docs claim to raise an error if you try to do both.

Comment (rfk) [inline test/api.js]: Aha, no, what used to get returned here was the result of `.done(done, done)` which is not a promise.

**Pull request merged**